hw/top_earlgrey/data/ip/chip_otp_ctrl_testplan.hjson

```diff
@@ -0,0 +1,227 @@
+{
+  name: chip_otp_ctrl
+  testpoints: [
+    // OTP (pre-verified IP) integration tests:
+    {
+      name: chip_otp_ctrl_init
+      desc: '''Verify the OTP ctrl initialization on chip power up.
+
+            Verify that the chip powers up correctly on POR.
+            - The pwrmgr initiates a handshake with OTP ctrl and later, with LC ctrl in subsequent
+              FSM states. Ensure that the whole power up sequence does not hang.
+            - Verify with connectivity assertion checks, the handshake signals are connected.
+            - Ensure that no interrupts or alerts are triggered.
+            '''
+      features: ["OTP_CTRL.INIT"]
+      stage: V2
+      si_stage: SV1
+      tests: ["chip_sw_lc_ctrl_transition"]
+    }
+    {
+      name: chip_sw_otp_ctrl_keys
+      desc: '''Verify the proliferation of keys to security peripherals.
+
+            - Verify the correctness of keys provided to SRAM ctrl (main & ret), flash ctrl, keymgr,
+              (note that keymgr does not have handshake).
+            - Ensure that the test requests a new key and verifies the previously written
+              data to an address now returns a garbage value.
+
+            X-ref'ed with the following IP tests that consume these signals:
+            - chip_sw_sram_scrambled_access
+            - chip_sw_flash_scramble
+            - chip_sw_keymgr_key_derivation
+            - chip_sw_otbn_mem_scramble
+            '''
+      features: ["OTP_CTRL.KEY_DERIVATION", "OTP_CTRL.PARTITION.SECRET1"]
+      stage: V2
+      si_stage: SV1
+      tests: [// Verifies both, main and retention SRAM scrambling.
+              "chip_sw_sram_ctrl_scrambled_access",
+              "chip_sw_flash_init",
+              "chip_sw_keymgr_key_derivation",
+              "chip_sw_otbn_mem_scramble",
+              "chip_sw_rv_core_ibex_icache_invalidate"]
+    }
+    {
+      name: chip_sw_otp_ctrl_entropy
+      desc: '''Verify the entropy interface from OTP ctrl to EDN.
+
+            This is X-ref'ed with the chip_otp_ctrl_keys test, which needs to handshake with the EDN
+            to receive some entropy bits before the keys for SRAM ctrl and OTBN are computed.
+            '''
+      features: ["OTP_CTRL.ENTROPY_READ"]
+      stage: V2
+      si_stage: SV1
+      tests: ["chip_sw_sram_ctrl_scrambled_access",
+              "chip_sw_flash_init",
+              "chip_sw_keymgr_key_derivation",
+              "chip_sw_otbn_mem_scramble",
+              "chip_sw_rv_core_ibex_icache_invalidate"]
+    }
+    {
+      name: chip_sw_otp_ctrl_program
+      desc: '''Verify the program request from lc_ctrl.
+
+            - Verify that upon an LC state transition request, LC ctrl signals the OTP ctrl with a
+              program request.
+            - Verify that the OTP ctrl generates the LC data output correctly and is sent to the LC
+              ctrl before it is reset.
+            - Verify that the `lc_check_byp_en_i` from LC ctrl is set.
```

Comment:
What I mean is that some checks like this and the one above cannot be done in the hardware: we can only check that the transition was done. I would encourage you to comment what can or cannot be checked in SiVal with a comment.

Reply:
I listed tests need to rewrite [here](https://github.com/lowRISC/opentitan/pull/19889#issue-1928567184)
I think how to revise it out of scope.